platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -184,6 +187,54 @@ class HangingProtocolService {
     this.protocols.set(protocolId, protocol);
   }
 
+  /** Sets the given protocol object as active.  Also sets the default
+   * for the given id to the specified protocol - allows updating which
+   * protocol is active for a given name.
+   */
+  public setActiveProtocol(id: string, protocol?: Protocol): void {
+    if (!id) {
+      throw new Error(`Protocol ${protocol} added without an id`);
+    }
+    console.log('Active protocol set', id, protocol);
+    if (protocol && this.protocols.get(id) !== protocol) {
+      this.addProtocol(id, protocol);
+    }
+    if (!this.activeProtocolIds) {
+      this.activeProtocolIds = [];
```

Comment:
Do you want to initiate this in the constructor instead of here?

Reply:
No, because null is used to indicate no specified hanging protocols, whereas this is used to specify only the specific set.